docs/package_templates/header_only/all/conanfile.py

```diff
@@ -57,22 +57,18 @@ def requirements(self):
     def package_id(self):
         self.info.clear()
 
-    @property
-    def _info(self):
-        return self if Version(conan_version).major < 2 else self.info
-
     def validate(self):
         # compiler subsettings are not available when building with self.info.clear()
-        if self._info.settings.compiler.get_safe("cppstd"):
+        if self.info.settings.compiler.get_safe("cppstd"):
```

Comment:
broken

Reply:
should it be `if self.info.settings.get_safe("compiler.cppstd"):` instead? @uilianries

Reply:
no, it should have the conan_version logic

Reply:
@prince-chrismc It does not work for Conan v1. It's same problem reported on https://github.com/conan-io/conan/issues/12210

So far, that workaround is our best option. Only `self.settings.` works for Conan v1, but on Conan v2 it's forbidden. We still don't have an answer about that problem.
Maybe adding a mock like beta5 will have for `self.env_info` ...

Reply:
I don't like conditional `conan_version` logic in recipes. IMO it should be possible to use the same code for v1/v2 for the migration. I would prefer `self.info` to have the same behavior in v1/v2. if it's impossible for some reason (why? we can break things in v2, right?), might we add `self.info2` with the consistent behavior for both version? or maybe we can add a new method like `pre_validate` that validates settings before they cleared from package ID?

Reply:
@SSE4 I don't like to see conditionals too, but so far we don't have a solution for that case. Conan 2.0 only works with `validate()` + `self.info.settings`, it raises an error if tries to use `self.settings` instead. And Conan v1 skips `cppstd` validation when building a header-only package, so for any package requiring C++17/20. As result, you will see the error only when building your project, because Conan v1 will not check it.

Reply:
> It does not work for Conan v1. It's same problem reported on [conan-io/conan#12210](https://github.com/conan-io/conan/issues/12210)

Yes exactly, we know that we dont know. So lets not add hacks that only work in one case while, this is slated for beta 5.
I've raised the issue within the team and I sure we will have a solution to move forward.

The whole point of hitting the migration in CCI early was to find problems like this. We do not need to solve them here -- it's a limitation in the client and/or a lack or understanding and/or a lack of clear docs.

***

This "solution" does not work for compiled projects since some of the helpers in 2.0 still use `self.settings` [check_min_vs](https://github.com/conan-io/conan/blob/d5d8c8279a1eb4c703e1c1efa7939468bfd1fc11/conan/tools/microsoft/visual.py#L20) for examples

So this just kicks the problem to every other template.